expression/builtin_string_vec.go

```diff
@@ -1770,11 +1770,71 @@ func (b *builtinBinSig) vecEvalString(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinFormatSig) vectorized() bool {
-	return false
+	return true
 }
 
+// evalString evals FORMAT(X,D).
+// See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_format
 func (b *builtinFormatSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	isDecimal := b.args[0].GetType().EvalType() == types.ETDecimal
+	var buf0 *chunk.Column
+	buf0, err := b.bufAllocator.get(b.args[0].GetType().EvalType(), n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	buf1, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf1); err != nil {
+		return err
+	}
+
+	xStrBuf := make([]string, n)
+	if isDecimal {
+		if err := b.args[0].VecEvalDecimal(b.ctx, input, buf0); err != nil {
+			return err
+		}
+		buf1.MergeNulls(buf0)
+		d64s := buf0.Decimals()
+		for i := 0; i < n; i++ {
+			if buf1.IsNull(i) {
+				continue
+			}
+			xStrBuf[i] = d64s[i].String()
+		}
+	} else {
+		if err := b.args[0].VecEvalReal(b.ctx, input, buf0); err != nil {
+			return err
+		}
+		buf1.MergeNulls(buf0)
+		f64s := buf0.Float64s()
+		for i := 0; i < n; i++ {
+			if buf1.IsNull(i) {
+				continue
+			}
+			xStrBuf[i] = fmt.Sprintf(strconv.FormatFloat(f64s[i], 'f', -1, 64))
+		}
+	}
```

Comment:
How about creating a new function for the logic like `evalNumDecArgsForFormat` then another format function `builtinFormatWithLocaleSig` can also use it.

Reply:
👌

Reply:
I will try it tomorrow.